pages/plan/index.js

```diff
@@ -103,39 +197,41 @@ export default function Plan({
 
       <CalendarContainer>
         {weekdays &&
-          weekdays.map((weekday, index) => (
-            <article key={weekday.date} id={weekday.date}>
-              <h2>{weekday.readableDate}</h2>
-              {weekday.recipe ? (
-                <MealCard
-                  key={weekday.recipe._id}
-                  recipe={weekday.recipe}
-                  isFavorite={getRecipeProperty(
-                    weekday.recipe._id,
-                    "isFavorite"
-                  )}
-                  onToggleIsFavorite={toggleIsFavorite}
-                />
-              ) : (
-                <CardSkeleton />
-              )}
-            </article>
-          ))}
-      </CalendarContainer>
-      <ButtonsContainer>
-        <GenerateButton
-          onClick={() => {
-            assignRecipesToWeekdays(
-              setWeekdays,
-              userRecipes,
-              randomRecipes,
-              numberOfRandomRecipes
+          weekdays.map((weekday) => {
+            const calendarDay = getCalendarDayFromDb(weekday.date);
+            return (
+              <article key={weekday.date} id={weekday.date}>
+                <h2>{weekday.readableDate}</h2>
+                {calendarDay?.recipe ? (
+                  <MealCard
+                    key={calendarDay.recipe._id}
+                    recipe={calendarDay.recipe}
+                    numberOfPeople={
+                      calendarDay.numberOfPeople !== undefined &&
+                      calendarDay.numberOfPeople !== null
+                        ? Number(calendarDay.numberOfPeople)
+                        : user.settings.defaultNumberOfPeople
+                    }
```

Comment:
This should work just as fine: 

```suggestion
                    numberOfPeople={Number(calendarDay.numberOfPeople) || user.settings.defaultNumberOfPeople}
```

Reply:
Thanks! I figured we don't need this prop here at all, so I removed it completely. :-)